### ISSUE TITLE
VSCode LSP doesn't work with Packaged Grain binaries

### ISSUE THREAD

**Issue by phated** [grain-lang/grain-language-server#44]
In grain-lang/grain#570, a binary packager was added to Grain; however, the VSCode extension's LSP doesn't work with them.

I have tracked the issue down to this line: https://github.com/grain-lang/grain-language-server/blob/master/editor-extensions/vscode/server/src/server.ts#L304 - it seems that when specifying `input` to node's `execFileSync` that it isn't actually flushing that input into the 0 file descriptor (stdin), but instead creating its own pipe and binding that to stdin.

I verified that normal stdin works using
```sh
$ ./grain lsp test.gr < test.gr
```

**Comment by phated** [comment]
We should try to figure this out by Wasm Summit on Thursday if possible!

**Comment by ng-marcus** [comment]
I'll take a look!

**Comment by ng-marcus** [comment]
I did some research.

Firstly, I tried using a different way of passing the stdin

`	var child = childProcess.spawn(cliPath, ['lsp', filename]); //the array is the arguments

					child.stdin.write(text); // send the file

					child.stdout.on('data', function (data) {
						console.log('stdout: ' + data);
					});
					child.stdin.end();`

But that failed in the same way.

The I replaced the program I called with this shell script

`#!/bin/sh
cat -`

which echoes stdin straight back, and that works with both methods.

So I'm starting to suspect it's the new pkg binary and how it's reading stdin.
